orca-echo/src/main/groovy/com/netflix/spinnaker/orca/echo/tasks/PageApplicationOwnerTask.groovy

```diff
@@ -42,32 +44,54 @@ class PageApplicationOwnerTask implements RetryableTask {
   @Autowired(required = false)
   Front50Service front50Service
 
+  private final ObjectMapper mapper = new ObjectMapper()
+
   @Override
   TaskResult execute(Stage stage) {
     if (!front50Service) {
       throw new UnsupportedOperationException("Front50 is not enabled, no way to fetch pager duty. Fix this by setting front50.enabled: true");
     }
-    def application = stage.context.application as String
-    def applicationPagerDutyKey = fetchApplicationPagerDutyKey(application)
-    if (!applicationPagerDutyKey) {
-      throw new IllegalStateException("${application} does not have a pager duty service key!")
+    def applications = stage.context.applications as List<String>
+    def keys = stage.context.keys as List<String>
```

Comment:
Would it be clearer if `context.keys` were `context.pagerDutyKeys`?

Either way you can do `def pagerDutyKeys = stage.context.keys as List<String>`

Reply:
Since it's a pager duty task, I think calling it pagerDutyKeys is superfluous, but totally agree on the `def pagerDutyKeys` part.